Jellyfin.Server.Implementations/Item/BaseItemRepository.cs

```diff
@@ -552,22 +552,32 @@ public ItemCounts GetItemCounts(InternalItemsQuery filter)
     }
 
     /// <inheritdoc  />
-    public void SaveImages(BaseItemDto item)
+    public async Task SaveImagesAsync(BaseItemDto item, CancellationToken cancellationToken = default)
     {
         ArgumentNullException.ThrowIfNull(item);
 
-        var images = item.ImageInfos.Select(e => Map(item.Id, e));
-        using var context = _dbProvider.CreateDbContext();
+        var images = item.ImageInfos.Select(e => Map(item.Id, e)).ToArray();
+
+        var context = await _dbProvider.CreateDbContextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Looks like this is missing a dispose

Reply:
Looks like when i try using, using, it gives me an error asking for me to ConfigureAwait and it seems to come down to the compiler expanding out for that Dispose and there doesnt seem to be a way for me to tell it to expand with a configure await on it so i manually expanded it out. How does that look?

Reply:
if this looks good i can squash

Reply:
Here's an example of the `await using` pattern
https://github.com/jellyfin/jellyfin/blob/45cb5a0008b27e326b8e14071ea03039dc6040c9/Jellyfin.Server.Implementations/Devices/DeviceManager.cs#L78-L79